src/pages/Facility/settings/organizations/FacilityOrganizationIndex.tsx

```diff
@@ -85,43 +92,216 @@ export default function FacilityOrganizationIndex({
       </Page>
     );
   }
+  const toggleRow = (id: string) => {
+    setExpandedRows((prev) => ({ ...prev, [id]: !prev[id] }));
+  };
+  const getChildren = (parentId: string) => {
+    return data.results.filter((org) => org.parent?.id === parentId);
+  };
+  const OrganizationRow = ({
+    org,
+    expandedRows,
+    toggleRow,
+    getChildren,
+    indent,
+  }: {
+    org: {
+      id: string;
+      name: string;
+      parent?: { id: string };
+      org_type: string;
+    };
+    expandedRows: Record<string, boolean>;
+    toggleRow: (id: string) => void;
+    getChildren: (parentId: string) => {
+      id: string;
+      name: string;
+      parent?: { id: string };
+      org_type: string;
+    }[];
+    indent: number;
+  }) => {
+    const children = getChildren(org.id);
+    const isTopLevel = !org.parent || Object.keys(org.parent).length === 0;
 
+    const toggleAllChildren = () => {
+      setExpandedRows((prevExpandedRows) => {
+        const newExpandedRows = { ...prevExpandedRows };
+        const toggleChildren = (parentId: string, expand: boolean) => {
+          getChildren(parentId).forEach((child) => {
+            newExpandedRows[child.id] = expand;
+            toggleChildren(child.id, expand);
+          });
+        };
+        const shouldExpand = !children.some(
+          (child) => prevExpandedRows[child.id],
+        );
+        newExpandedRows[org.id] = shouldExpand;
+        toggleChildren(org.id, shouldExpand);
+        return newExpandedRows;
+      });
+    };
+    return (
+      <>
+        <TableRow
+          key={org.id}
+          style={{ "--indent": `${indent * 20}px` } as React.CSSProperties}
+        >
+          <TableCell
+            className={`${
+              isTopLevel
+                ? "bg-white font-bold text-[#030712]"
+                : "bg-white font-medium text-[#030712]"
+            } flex justify-between items-center pl-[var(--indent)] flex-wrap gap-2`}
+          >
+            <div className="flex items-center">
+              {children.length > 0 ? (
+                <button onClick={() => toggleRow(org.id)} className=" mr-1">
+                  {expandedRows[org.id] ? (
+                    <CareIcon icon="l-angle-down" className="h-5 w-5" />
+                  ) : (
+                    <CareIcon icon="l-angle-right" className="h-5 w-5" />
+                  )}
+                </button>
+              ) : org.parent && Object.keys(org.parent).length > 0 ? (
+                <CareIcon
+                  icon="l-corner-up-right"
+                  className="h-4 w-4 text-gray-400"
+                />
+              ) : null}
+              {org.name}
+            </div>
+            {isTopLevel && (
+              <>
+                {children.length > 0 ? (
+                  <>
+                    <Button
+                      variant="outline"
+                      className="h-7 shadow-gray-400 border-gray-400 sm:p-2 sm:text-sm text-xs p-1"
```

Comment:
Let's override styles, instead write a new variant if necessary.

In this case, white variant fits (not same as figma, but bit more subtle).

Reply:
try xs or sm size for expand/collapse button.